File: 0001-two-sum/0001-two-sum.py
from collections import defaultdict
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        sums = defaultdict(int)
        for i in range(n):
            diff = target - nums[i]
            if diff in sums:
                return [sums[diff], i]
            else:
                sums[nums[i]] = i
        

        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        Map<Integer, Integer> seen = new HashMap<>();

        for (int i=0; i<nums.length; i++){
            int n = nums[i];

            if (seen.containsKey(target-n)){
                return new int[]{i, seen.get(target-n)};
            }
            seen.put(n, i);
        }
        return new int[]{};
        
    }
}